### PR TITLE
Docs: fix broken links in Node.js API docs toc

### PR DESCRIPTION
**Pull request by u-sho**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

I was reading the Node.js API page and I noticed that the links in the table of contents don't work in the website. This PR only remove hyphen from the link references to fix those links in the website.

I'm aware that currently there is the opened PR #13415 but I notice that there wasn't any activity for about 9 months.
Since I found this issue and probably other people will stumble on it, I thought to go ahead and open this PR with a smaller scope to speed the push of the fix like #13771 (and because of my faltering English, I borrowed comment from the PR)

#### Is there anything you'd like reviewers to focus on?

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

**Comment by mdjermanovic** [comment]
@u-sho thanks for the PR!

Can you please sign our [CLA](https://cla.js.foundation/eslint/eslint?pullRequest=14296), so we could merge this?

File: docs/developer-guide/nodejs-api.md [hunk 1]
```diff
@@ -21,15 +21,15 @@ While ESLint is designed to be run on the command line, it's possible to use ESL
...
+    * [verify()](#linterverify)
+    * [verifyAndFix()](#linterverifyandfix)
+    * [defineRule()](#linterdefinerule)
     * [defineRules()](#linter-definerules)
-    * [getRules()](#linter-getrules)
-    * [defineParser()](#linter-defineparser)
-    * [version](#linter-version)
+    * [getRules()](#lintergetrules)
+    * [defineParser()](#linterdefineparser)
+    * [version](#linterversionlinterversion)
```
Comment (mdjermanovic) [inline docs/developer-guide/nodejs-api.md:32]: `#linter-definerules` should be fixed, too?

Comment (u-sho) [reply docs/developer-guide/nodejs-api.md:32]: It's my mistake. I will fix soon.

**Comment by u-sho** [comment]
@mdjermanovic I appreciate your review. I fixed what you commented and signed the
 CLA. And I'll never mind that this PR will be merged.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Comment by mdjermanovic** [comment]
Thanks for contributing!

**Pull request merged**